src/s390_ecc.c

```diff
@@ -468,7 +468,8 @@ unsigned int make_ecdsa_verify_parmblock(char *pb,
  * makes an ECDSA key structure at given struct and returns its length.
  */
 unsigned int make_ecdsa_private_key_token(unsigned char *kb,
-		const ICA_EC_KEY *privkey, uint8_t curve_type)
+		const ICA_EC_KEY *privkey, unsigned char *X, unsigned char *Y,
+		uint8_t curve_type)
```

Comment:
is there a reason to restrict the curve type to 8bit. why not just use an int?

Reply:
Yes, there is a reason .... the curve_type field in the ECC_PRIVATE_KEY_SECTION is a 8-bit value: 
kp1->privsec.curve_type = curve_type;
If curve_type has already the right size, we don't need a cast. In fact I'm "enforcing" this cast by passing the curve_type parm to make_ecdsa_private_key_token(). In the calling routine make_ecdsa_sign_request() curve_type is in fact a int. But I could change the curve_type_from_nid() routine to return a uint8_t instead of an int, so that we have always the right type.

Reply:
okay. if the curve_type is a 8-bit exact-width int, id suggest to use it uint8_t consistently.

Reply:
I generally I vote for int or unsigned int here and just doing the cast the moment you fill the CPRB struct.
This way the API has room for future extensions, it is narrowed down exactly where the one and only
hardware is used. But new hardware (e.g. ECC on CPACF) may not have this restrictions and may support
future ECC curves...